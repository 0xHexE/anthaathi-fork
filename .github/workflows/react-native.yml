name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
