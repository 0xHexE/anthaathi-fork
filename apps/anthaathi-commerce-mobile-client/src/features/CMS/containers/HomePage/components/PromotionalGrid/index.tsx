import * as React from 'react';
import {CommonPlacement, HomePageComponentType} from '../../../../types/common';
import {GestureResponderEvent, Image, View} from 'react-native';
import {Text, TouchableRipple, useTheme} from 'react-native-paper';
import {MD3Colors} from 'react-native-paper/lib/typescript/types';
import {
  ResponsiveInput,
  useResponsiveValue,
} from '../../../../utils/useResponsiveValue';

export interface PromotionalGridProps {
  items: PromotionalGridItem[];
}

export interface PromotionalGridItem {
  subHeading: string;
  heading: string;
  text: string;
  button1Text: string;
  onPress1?: () => void;
  button2Text?: string;
  onPress2?: () => void;
  image: string;
  videoURL?: string;
  width?: ResponsiveInput<'33%' | '50%' | '100%'>;
  height: ResponsiveInput<number>;
  textAlignment?: CommonPlacement;
  textColor?: string;
  key: string;
  paddingLeft?: ResponsiveInput<number>;
  paddingRight?: ResponsiveInput<number>;
  paddingBottom?: ResponsiveInput<number>;
  paddingTop?: ResponsiveInput<number>;
}

export default function PromotionalGrid(props: PromotionalGridProps) {
  return (
    <View
      style={{
        display: 'flex',
        flexDirection: 'row',
        flexWrap: 'wrap',
      }}
      testID="promotionalGrid">
      {props.items.map(item => {
        return <PromotionalGridItemRenderer key={item.key} item={item} />;
      })}
    </View>
  );
}

/* A Component that returns a button. */
function PromotionalGridButton(props: {
  onPress: ((e: GestureResponderEvent) => void) | undefined;
  label: string;
}) {
  const theme = useTheme();

  return (
    <TouchableRipple onPress={props.onPress}>
      <View style={{backgroundColor: (theme.colors as MD3Colors).background}}>
        <Text
          style={{
            color: (theme.colors as MD3Colors).onBackground,
            paddingLeft: 18,
            paddingRight: 18,
            paddingTop: 8,
            paddingBottom: 8,
            fontWeight: 'bold',
          }}
          variant="labelSmall">
          {props.label}
        </Text>
      </View>
    </TouchableRipple>
  );
}

function PromotionalGridItemRenderer({item}: {item: PromotionalGridItem}) {
  const theme = useTheme();

  const itemWidth = useResponsiveValue(item.width ?? '33%');
  const itemHeight = useResponsiveValue(item.height);
  const paddingLeft = useResponsiveValue(item.paddingLeft ?? 12);
  const paddingRight = useResponsiveValue(item.paddingRight ?? 12);
  const paddingTop = useResponsiveValue(item.paddingTop ?? 12);
  const paddingBottom = useResponsiveValue(item.paddingBottom ?? 12);

  return (
    <View
      style={{
        width: itemWidth,
        paddingLeft: paddingLeft,
        paddingRight: paddingRight,
        paddingTop: paddingTop,
        paddingBottom: paddingBottom,
      }}>
      <View
        style={{
          height: itemHeight,
        }}>
        <Image
          source={{
            uri: item.image,
          }}
          style={{
            position: 'absolute',
            height: '100%' as never,
            width: '100%' as never,
          }}
        />

        <View
          style={{
            flexGrow: 1,
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center',
          }}>
          <Text
            style={{
              textAlign: item.textAlignment ?? 'center',
              color: (theme.colors as MD3Colors).onPrimary,
              marginBottom: 6,
            }}
            variant="titleSmall">
            {item.subHeading}
          </Text>
          <Text
            style={{
              textAlign: item.textAlignment ?? 'center',
              color: (theme.colors as MD3Colors).onPrimary,
              marginBottom: 6,
            }}
            variant="titleLarge">
            {item.heading}
          </Text>
          <Text
            style={{
              textAlign: item.textAlignment ?? 'center',
              color: (theme.colors as MD3Colors).onPrimary,
              marginBottom: 12,
            }}
            variant="bodyMedium">
            {item.text}
          </Text>
        </View>

        {(item.button1Text || item.button2Text) && (
          <View
            style={{
              display: 'flex',
              flexDirection: 'row',
              alignItems: 'center',
              justifyContent: 'center',
              paddingBottom: 12,
            }}>
            {item.button1Text && (
              <PromotionalGridButton
                label={item.button1Text}
                onPress={item.onPress1}
              />
            )}
            {item.button2Text && (
              <>
                <View style={{width: 12}} />
                <PromotionalGridButton
                  label={item.button2Text}
                  onPress={item.onPress1}
                />
              </>
            )}
          </View>
        )}
      </View>
    </View>
  );
}

export const PromotionalGridCMSInput = {
  _component: HomePageComponentType.PromotionalGrid,
  component: PromotionalGrid,
};
