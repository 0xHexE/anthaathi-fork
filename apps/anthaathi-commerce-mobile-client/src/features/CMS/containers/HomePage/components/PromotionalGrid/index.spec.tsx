import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import PromotionalGrid from './index';

describe('PromotionalGrid', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <PromotionalGrid
            items={[
              {
                key: '12',
                subHeading: 'Something',
                heading: 'Something',
                text: 'test',
                button1Text: 'SHOP NOW',
                height: [180, 240, 260, 270],
                image:
                  'https://burst.shopifycdn.com/photos/fruit-plate.jpg?width=373&height=373&format=pjpg&exif=1&iptc=1',
                width: ['100%', '50%', '100%', '100%'],
              },
              {
                key: '1233',
                subHeading: 'Something',
                heading: 'Something',
                text: 'test',
                button1Text: 'SHOP NOW',
                height: [180, 240, 260, 270],
                image:
                  'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
                width: ['100%', '50%', '100%', '100%'],
              },
            ]}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('promotionalGrid')).toBeTruthy();
  });
});
