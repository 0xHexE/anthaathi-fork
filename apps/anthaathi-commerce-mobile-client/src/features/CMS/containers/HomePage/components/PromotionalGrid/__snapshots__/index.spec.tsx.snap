// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromotionalGrid should render the item 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
  testID="promotionalGrid"
>
  <View
    style={
      Object {
        "paddingBottom": 12,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 12,
        "width": "50%",
      }
    }
  >
    <View
      style={
        Object {
          "height": 240,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://burst.shopifycdn.com/photos/fruit-plate.jpg?width=373&height=373&format=pjpg&exif=1&iptc=1",
          }
        }
        style={
          Object {
            "height": "100%",
            "position": "absolute",
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexGrow": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "marginBottom": 6,
                "textAlign": "center",
              },
            ]
          }
        >
          Something
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontSize": 22,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 28,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "marginBottom": 6,
                "textAlign": "center",
              },
            ]
          }
        >
          Something
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontSize": 14,
                "fontWeight": "400",
                "letterSpacing": 0.25,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "marginBottom": 12,
                "textAlign": "center",
              },
            ]
          }
        >
          test
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 12,
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              false,
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(255, 251, 254, 1)",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontSize": 11,
                    "fontWeight": "500",
                    "letterSpacing": 0.5,
                    "lineHeight": 16,
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontWeight": "bold",
                    "paddingBottom": 8,
                    "paddingLeft": 18,
                    "paddingRight": 18,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              SHOP NOW
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingBottom": 12,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 12,
        "width": "50%",
      }
    }
  >
    <View
      style={
        Object {
          "height": 240,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1",
          }
        }
        style={
          Object {
            "height": "100%",
            "position": "absolute",
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexGrow": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "marginBottom": 6,
                "textAlign": "center",
              },
            ]
          }
        >
          Something
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontSize": 22,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 28,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "marginBottom": 6,
                "textAlign": "center",
              },
            ]
          }
        >
          Something
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontSize": 14,
                "fontWeight": "400",
                "letterSpacing": 0.25,
                "lineHeight": 20,
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "marginBottom": 12,
                "textAlign": "center",
              },
            ]
          }
        >
          test
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 12,
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              false,
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(255, 251, 254, 1)",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontSize": 11,
                    "fontWeight": "500",
                    "letterSpacing": 0.5,
                    "lineHeight": 16,
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontWeight": "bold",
                    "paddingBottom": 8,
                    "paddingLeft": 18,
                    "paddingRight": 18,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              SHOP NOW
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
